### PR TITLE
Browsing support for code coverage

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Testing : manual and unit tests. Will add this verification in E2E tests too.

### CODE REVIEW

File: src/agent/codecoveragepublisher.ts [hunk 1]
```diff
@@ -118,7 +118,7 @@ export class CodeCoveragePublisher {
                 var rawFilesArtifactName = "Code Coverage Files_" + _this.buildId;
 
                 _this.command.info("PublishCodeCoverageFiles : Publishing additional code coverage files '" + rawFilesDirectory + "'");
-                _this.uploadArtifact(rawFilesDirectory, rawFilesArtifactName, containerId).then(function() {
+                _this.uploadArtifact(rawFilesDirectory, rawFilesArtifactName, containerId, "False").then(function() {
```
Comment (nigurr) [inline src/agent/codecoveragepublisher.ts]: use default type boolean false instead of string

File: src/agent/codecoveragepublisher.ts [hunk 2]
```diff
@@ -201,6 +204,17 @@ export class CodeCoveragePublisher {
         return defer.promise;
     }
     
+    //-----------------------------------------------------
+    // Finds if the report directory has index.html
+    // - reportDirectory: string  - report directory 
+    //-----------------------------------------------------    
+    private isReportDirectoryBrowsable(reportDirectory: string): string {
+        var defaultIndexFile = path.join(reportDirectory, "index.html");
+        if(utilities.isPathExists(defaultIndexFile) && fs.lstatSync(defaultIndexFile).isFile()){
```
Comment (nigurr) [inline src/agent/codecoveragepublisher.ts]: return boolean

File: src/test/lib/feedback.ts [hunk 3]
```diff
@@ -174,8 +176,12 @@ export class TestFeedbackChannel extends events.EventEmitter implements cm.IServ
 
     public postArtifact(projectId: string, buildId: number, artifact: buildifm.BuildArtifact): Q.Promise<buildifm.BuildArtifact> {
         this.artifactNames.push(artifact.name);
-        if(this.failingArtifactName == artifact.name)
-        {
+        
+        if (artifact.resource.properties["browsable"] == "True") {
```
Comment (nigurr) [inline src/test/lib/feedback.ts]: ignore case while comapring

File: src/agent/codecoveragepublisher.ts (hunk 2 above)
Comment (nigurr) [inline src/agent/codecoveragepublisher.ts]: lstatsync actually verifies if path exists or not.

**Comment by nigurr** [comment]
Approved with Suggestions

File: src/test/lib/feedback.ts (hunk 3 above)
Comment (AshwiniChalla) [inline src/test/lib/feedback.ts]: We are particular about casing too.

File: src/agent/codecoveragepublisher.ts (hunk 2 above)
Comment (AshwiniChalla) [inline src/agent/codecoveragepublisher.ts]: It can also be used to check if path i s file or not.

Comment (AshwiniChalla) [inline src/agent/codecoveragepublisher.ts]: resolved.

File: src/agent/codecoveragepublisher.ts (hunk 1 above)
Comment (AshwiniChalla) [inline src/agent/codecoveragepublisher.ts]: string conversion does it to true but server side code expects True

**Pull request merged**